src/cmd/cmd_utils.ml

```diff
@@ -38,6 +38,12 @@ let write_testcase =
     in
     res
 
+let tmp_dir pat =
+  match Bos.OS.Dir.tmp pat with
+  | Ok path -> path
+  | Error e ->
+    Fmt.failwith "Failed to create tmp directory: %a" Rresult.R.pp_msg e
```

Comment:
Can you avoid using `Rresult`? Also, I believe you could simply use:

```ocaml
let* path = Bos.OS.Dir tmp pat
```

Everywhere instead of declaring this one ?

Reply:
Using it in `~default:` will still create it (~default is not lazy)